src/contract/deploy.rs

```diff
@@ -46,17 +48,26 @@ impl<T: Transport> Builder<T> {
     pub fn execute<P, V>(self, code: V, params: P, from: Address) -> Result<PendingContract<T>, ethabi::Error>
     where
         P: Tokenize,
-        V: Into<Vec<u8>>,
+        V: AsRef<str>,
     {
         let options = self.options;
         let eth = self.eth;
         let abi = self.abi;
 
+        let mut code_hex = code.as_ref().to_string();
+
+        for (lib, address) in self.linker {
+            let replace = format!("__{:_<1$}", lib, 38-lib.len());
```

Comment:
That might underflow if `lib.len() > 38` I'd rather have an explicit error if that's the case